validate/validate.go

```diff
@@ -423,25 +396,30 @@ func (v *Validator) CheckMounts() (msgs []string) {
 	return
 }
 
-// CheckOS checks v.spec.Platform.OS
-func (v *Validator) CheckOS() (msgs []string) {
-	logrus.Debugf("check os")
+// CheckPlatform checks v.platform
+func (v *Validator) CheckPlatform() (msgs []string) {
+	logrus.Debugf("check platform")
+
+	if v.platform != "linux" && v.platform != "solaris" && v.platform != "windows" {
```

Comment:
I'm fine with either, if check platform right after got it from context, it would be like a runtime-tools limitation, while it's actually can be an OCI validation, so check it in `Validator` methods seems reasonable to be.

Plus I moved `CheckPlatform` to the top in the second commit, so we don't do much unnecessary check it platform is wrong.